swap/swap.go

```diff
@@ -168,17 +191,29 @@ func (s *Swap) updateBalance(peer enode.ID, amount int64) (int64, error) {
 	s.balances[peer] += amount
 	//save the new balance to the state store
 	peerBalance := s.balances[peer]
-	err := s.stateStore.Put(peer.String(), &peerBalance)
+	err := s.stateStore.Put(balanceKey(peer), &peerBalance)
 	if err != nil {
 		log.Error("error while storing balance for peer", "peer", peer.String())
 	}
 	log.Debug("balance for peer after accounting", "peer", peer.String(), "balance", strconv.FormatInt(peerBalance, 10))
 	return peerBalance, err
 }
 
+// loadBalance loads balances from the state store (persisted)
+func (s *Swap) loadBalance(peer enode.ID) (err error) {
```

Comment:
this function should be then called something along the lines of `lazyLoadBalance`

Reply:
This name should be consistent with the other similar functions–and we have a `loadLastReceivedCheque` function which is comparable to this one.

If all of the names should be changed, I suggest we do this on a separate issue.